ksql-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -76,42 +80,41 @@ public PlanNode buildPlan() {
       currentNode = buildProjectNode(currentNode.getSchema(), currentNode);
     }
 
-    return buildOutputNode(currentNode.getSchema(), currentNode);
+    return buildOutputNode(analysis.getFromDataSource(0).left.getSchema(),
+        currentNode.getSchema(),
+        currentNode);
   }
 
-  private OutputNode buildOutputNode(final Schema inputSchema, final PlanNode sourcePlanNode) {
+  private OutputNode buildOutputNode(final Schema sourceSchema,
+                                     final Schema inputSchema,
```

Comment:
What is the difference between `sourceSchema` and `inputSchema`?
Is `sourceSchema` the same as `sourcePlanNode.getSchema()`?

Reply:
`sourceSchema` is the one from the source datasource, i.e., `analysis.getFromDataSource()`